File: test/resources/HelloProphecy/prophecy_scala/pipelines/raw_bronze/code/src/main/scala/io/prophecy/pipelines/raw_bronze/graph/ReformatOrders.scala
package io.prophecy.pipelines.raw_bronze.graph

import io.prophecy.libs._
import io.prophecy.pipelines.raw_bronze.udfs.PipelineInitCode._
import io.prophecy.pipelines.raw_bronze.udfs.UDFs._
import io.prophecy.pipelines.raw_bronze.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object ReformatOrders {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.select(
      col("order_id").cast(IntegerType).as("order_id"),
      col("customer_id").cast(IntegerType).as("customer_id"),
      col("order_status"),
      col("order_category"),
      col("order_date").cast(DateType).as("order_date"),
      col("amount").cast(FloatType).as("amount")
    )

}

File: test/resources/HelloProphecy/prophecy/pipelines/raw_bronze/code/raw_bronze/graph/ReformatOrders.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from raw_bronze.config.ConfigStore import *
from raw_bronze.udfs.UDFs import *

def ReformatOrders(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0.select(
        col("order_id").cast(IntegerType()).alias("order_id"), 
        col("customer_id").cast(IntegerType()).alias("customer_id"), 
        col("order_status"), 
        col("order_category"), 
        col("order_date").cast(DateType()).alias("order_date"), 
        col("amount").cast(FloatType()).alias("amount")
    )
